src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+    verify:
+        N addresses created
+        Contact address matches address object address
+    **********************************************************************************************************/            
+    @isTest
+    public static void newRecordsWithAddrInfo() {            
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Contacts_Addresses_Enabled__c = true));
+ 
+        Integer contactCount = 3;
+        List<Contact> contacts = UTIL_UnitTestData_TEST.createMultipleTestContacts(contactCount);
+        for (Contact contact : contacts) {
+            contact.MailingStreet = '123 45th';
+            contact.MailingCity = 'Seattle';
+        }
+        Test.startTest();
+        insert contacts;
+        Test.stopTest();
+
+        // verify results
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        system.assertEquals(contactCount, mapAccIdAcc.size());
+        
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, MailingCity__c, Parent_Contact__c, 
+        Latest_Start_Date__c, Latest_End_Date__c from Address__c];
+        system.assertEquals(contactCount, ListAddr.size());
+        
+        for (Address__c addr : ListAddr) {
+            Contact contact = mapAccIdAcc.get(addr.Parent_Contact__c);
+            system.assertEquals(contact.MailingStreet, addr.MailingStreet__c);
+            system.assertEquals(contact.MailingCity, addr.MailingCity__c);
+            system.assertEquals(true, addr.Default_Address__c);
+        }
+    }
+        
+    /*********************************************************************************************************
+    @description Update existing default Addresses.  
+    verify: Contact Mailing address fields updated
+    **********************************************************************************************************/            
+    @isTest 
+    public static void updateDefaultAddr() {
+        configSettings();
+
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // now let's update the Addresses
+        for (Integer i = 0; i < contactsAddrs.addrs.size(); i++) {
+            Address__c addr = contactsAddrs.addrs[i];
+            addr.MailingStreet__c = 'New Street' + i;
+            addr.MailingCity__c = 'New City' + i;
+        }
+        system.assertEquals(false, ADDR_Addresses_TDTM.hasRunAddrTrigger);
+        Test.startTest();
+        update contactsAddrs.addrs;
+        Test.stopTest();
+    
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause or contain the number of records redurned, or adding a `LIMIT` clause to cap the maximum number of records returned.